.circleci/config.yml

```diff
@@ -1,23 +1,34 @@
 version: 2.1
 
 orbs:
-  python: circleci/python@0.2.1
+  python: circleci/python@2.1.1
+
 jobs:
   install_and_update_dependencies:
     docker:
-      - image: circleci/python:3.8.6-node
+      - image: cimg/python:3.8.6-node
     working_directory: ~/repo
     steps:
       - checkout
       - restore_cache:
           keys:
-            - v2-dependencies-{{ checksum "Pipfile.lock" }}-{{ checksum "thumbs/Pipfile.lock" }}
-            - v2-dependencies-default
-      - run: &install-pipenv sudo pip install -U urllib3 cryptography==3.3.2 pipenv pip --quiet --no-input
-      - run: sudo pip uninstall -y poetry
-      - run: pipenv install --dev
-      - run: pipenv check # before save_cache so an insecure cache is never saved
+            - v3-dependencies-{{ checksum "Pipfile.lock" }}-{{ checksum "thumbs/Pipfile.lock" }}
+            - v3-dependencies-default
+      # Ensure pip and pipenv are installed
+      - run:
+          name: Install pipenv
+          command: |
+            python -m pip install --upgrade pip urllib3 cryptography==3.3.2 --quiet --no-input
+            pip install pipenv
+      - run: pip uninstall -y poetry
+      # Ensure that pipenv will create the virtual environment in the project directory
+      - run:
+          name: Set up virtual environment
+          command: |
+            export PIPENV_VENV_IN_PROJECT=1
```

Comment:
This step was needed to explicitly set up the virtual env following several failures to do so with `pipenv install --dev`.

Reply:
Can you explain what `PIPENV_VENV_IN_PROJECT` is needed for, or what problem it solves?
I pushed a branch at
https://github.com/DemocracyClub/electionleaflets/commits/251-review/
with one commit that just removes this, and it worked fine.

I wonder if one the the things this actually "fixed" was basically busting the cache when the lockfile hash hadn't changed because you're moving the virtualenv to a non-cached location?

Reply:
I added this early on to address missing venv errors in CI. I've now removed it.